UofS-Pulse-Binfo/kp_searches
# Germplasm Searches

Provides the following searches:
 - Germplasm: `search/germplasm/all`
 - Breeding Crosses: `search/germplasm/crosses`
 - Registered Varieties: `search/germplasm/varieties`
 - RILs: `search/germplasm/RILs`

## Setup
(done on Fresh)
1. Install or update `chado_search_api` and `kp_searches`. Make sure kp_searches is on this branch.
2. Update `kp_views`. Got to Structure > Views and 
  - disable the following: 
<img width="1619" alt="Screen Shot 2019-08-22 at 12 26 38 PM" src="https://user-images.githubusercontent.com/1566301/63539798-1ffa7b00-c4d8-11e9-9b54-4baef853ac5f.png">

  - delete/disable `F1_search`, `Recombinant_Inbred_Line_search`, `Registered_Variety_search`

3. Clear the cache using `drush cc all` and update the option cache by running `drush php-eval "kp_searches_cache_options();"`
4. Update menu items under Germplasm to point to the new searches.

# Testing
1. Check all the above searches display without errors.
2. Submit search without any filter criteria: check speed and 1st page of results
3. Check pager
4. Check various combinations of filter criteria
5. Ensure that when logged out you can only see the varieties and germplasm search. Also in the germplasm search all results should be of type variety or accession.
6. Check that links to content work.
7. Evaluate general ease of using the searches and if they are intuitive.
